src/main/java/com/craftaro/epicfarming/EpicFarming.java

```diff
@@ -256,6 +256,23 @@ public void onDataLoad() {
 
             this.farmManager.addFarms(this.dataManager.loadBatch(Farm.class, "active_farms"));
             this.boostManager.addBoosts(this.dataManager.loadBatch(BoostData.class, "boosted_players"));
+
+        });
+        Bukkit.getScheduler().runTaskAsynchronously(this, () -> {
```

Comment:
I am a bit concerned about potential data loading race conditions/inconsistencies.
(A common problem with Songoda plugins ^^)

There is another `#runTaskAsynchronously` that is involved in loading data (legacy-data-migration + `#addFarms` and `#addBoosts`).

Did you make sure this is not an issue? As both async runs can run in parallel and don't have to run sequentially. A Farm might be loaded in the new async task but overridden by the async-task above.
Especially with bigger datasets.

I would suggest merging the two async tasks into one.

Reply:
This has been done